src/contexts/Api/defaults.ts

```diff
@@ -9,6 +9,34 @@ import {
   APIContextInterface,
   ConnectionStatus,
 } from 'contexts/Api/types';
+import { NetworkName } from 'types';
+
+/**
+ * Checking values of "NETWORKS" instead of "NetworkName", because they're not
+ * compatible between runtime (the former) and compile time (the latter), the former
+ * having less fields defined and being the only safe one to use. This is obviously
+ * a wrong types problem, but fixing these types requires a bigger refactor.
+ */
+const isValidConfiguredNetworkName = (value: unknown): value is NetworkName =>
+  Object.keys(NETWORKS).includes(value as string);
+
+const defaultNetworkName =
+  process.env.NODE_ENV === 'production' &&
+  isValidConfiguredNetworkName(NetworkName.AlephZero)
+    ? NetworkName.AlephZero
+    : NetworkName.AlephZeroTestnet;
+
+const cachedNetworkName = localStorage.getItem('network');
+
+if (!isValidConfiguredNetworkName(cachedNetworkName)) {
+  localStorage.setItem('network', defaultNetworkName);
+}
+
+export const initialNetworkName = isValidConfiguredNetworkName(
+  cachedNetworkName
+)
+  ? cachedNetworkName
+  : defaultNetworkName;
```

Comment:
perhaps reversing order of this check would make it clearer:
```suggestion
export const initialNetworkName = isValidConfiguredNetworkName(
  cachedNetworkName
)
  ? cachedNetworkName
  : defaultNetworkName;
 
if (cachedNetworkName !== initialNetworkName) {
  localStorage.setItem('network', initialNetworkName);
}
```

Reply:
All right